Formula/u/uhdm.rb

```diff
@@ -0,0 +1,76 @@
+class Uhdm < Formula
+  include Language::Python::Virtualenv
+
+  desc "Universal Hardware Data Model, modeling of the SystemVerilog Object Model"
+  homepage "https://github.com/chipsalliance/UHDM"
+  url "https://github.com/chipsalliance/UHDM/archive/refs/tags/v1.71.tar.gz"
+  sha256 "3aa3783bc3fd5054bfe4347de1e2edfaa1bb3fbfbecdc3232413e826e95113c7"
+  license "Apache-2.0"
+  head "https://github.com/chipsalliance/UHDM.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "python@3.11" => :build
+  depends_on "six" => :build
+  depends_on "pkg-config" => :test
+  depends_on "capnp"
+
+  resource "orderedmultidict" do
+    url "https://files.pythonhosted.org/packages/53/4e/3823a27d764bb8388711f4cb6f24e58453e92d6928f4163fdb01e3a3789f/orderedmultidict-1.0.1.tar.gz"
+    sha256 "04070bbb5e87291cc9bfa51df413677faf2141c73c61d2a5f7b26bea3cd882ad"
+  end
+
+  def install
+    venv = virtualenv_create(buildpath/"venv", "python3.11")
+    resources.each do |r|
+      venv.pip_install r
+    end
+    # Build shared library
+    system "cmake", "-S", ".", "-B", "build_shared",
+      "-DBUILD_SHARED_LIBS=ON",
+      "-DUHDM_BUILD_TESTS=OFF",
+      "-DUHDM_USE_HOST_GTEST=ON",
+      "-DUHDM_USE_HOST_CAPNP=ON",
+      "-DCMAKE_INSTALL_RPATH=#{rpath}",
+      "-DCMAKE_CXX_FLAGS=-fPIC",
```

Comment:
@homebrew/core opinions?

Reply:
Would also like to know why this is needed and unsure that our shims will respect this anyway.

Reply:
which part is needed? 31 and 32 are flags to the library, 33 is standard from brew, and 34 is to link against.

Reply:
Why `-fPIC` is needed.

> 33 and 34 shouldn't really be necessary but I just copied from existing formula and upstream CI/CD build

Don't bother then unless it's known broken otherwise, thanks.

Reply:
Sure, I will copy this pattern for the ~5 other libraries I need to get to the one I actually wanted 😅, so hopefully this file looks good as a general pattern